### ISSUE TITLE
blazor-boot not path aware

### ISSUE THREAD

**Issue by torhovland** [aspnet/Blazor#360]
It looks like there is an issue with blazor-boot when you start the app from a route that is not the top-level. At least this is the case when running a standalone Blazor app at v0.1.0.

For example, if I access the app at https://myapp/products/123, this part of index.html

```
<script type="blazor-boot"></script>
```

gets turned into the relative path

```
<script src="_framework/blazor.js" ...
```

which means that the browser tries to load https://myapp/products/_framework/blazor.js, which fails.

**Comment by rstropek** [comment]
No, it should not. If you [configure the web server for a Single Page App (SPA)](https://learn-blazor.com/getting-started/static-hosting/), the request `https://myapp/products/123` will return *index.html*. It contains a `<base href="/">` element. The browser will use this base URL for all relative paths in the document: *The HTML <base> element specifies the base URL to use for all relative URLs contained within a document* ([source](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/base)).

Does this help?

**Comment by torhovland** [comment]
Hi, @danroth27 !

The reason I didn't have it is because I'm building a Blazor version of https://realworld.io/, and they supply some HTML as a starting point. I've added blazor-boot to that, of course, but I missed the `    <base href="/" />`. 

Anyway, you can see the work in progress here:
https://blazor-realworld.azurewebsites.net
https://github.com/torhovland/blazor-realworld-example-app

**Comment by danroth27** [comment]
> I'm building a Blazor version of https://realworld.io/

@torhovland That's fantastic! We'd love to hear more about your experience and the issues you run into.

**Comment by torhovland** [comment]
@danroth27 Sure! To be honest, I expected a lot of issues, e.g. with REST API calls with HTTP headers, JS interop, rendering bugs, etc., but in reality everything has been going fairly smoothly, and Blazor is already at the point where it feels more productive than JS frameworks like React and Angular, at least to me.

The issues I've run into are:
- the lag in VS 15.7 when editing Razor fragments.
- The JSON handling in HttpClient seems case sensitive, so if the JSON attributes are lowercase, then my class properties need to be lowercase as well.
- a NuGet package (Humanizer) that wouldn't work, not 100% sure if it's because of Blazor.
